### PR TITLE
Support uniform api and sigmoid api in new AD

### PR DESCRIPTION
**Pull request by Charles-hit**
### PR types

 New features
### PR changes

OPs
### Describe

Add `uniform_random_p` autograd primitive op and support orig2prim for paddle original `uniform_random` op and `sigmoid`op.

### CODE REVIEW

File: python/paddle/incubate/autograd/primrules.py [hunk 1]
```diff
@@ -464,6 +465,38 @@ def dropout_orig2prim(op, seed_t, x):
         )
 
 
+@REGISTER_ORIG2PRIM('uniform_random')
+def uniform_random_orig2prim(op, shape_t, shape_tl):
+    min_value = op.attr('min')
+    max_value = op.attr('max')
+    seed = op.attr('seed')
+    dtype = convert_np_dtype_to_dtype_(
+        convert_dtype(INT_DTYPE_2_STRING[op.attr('dtype')]))
```
Comment (cxxly) [inline python/paddle/incubate/autograd/primrules.py]: paddle.dtype(op.attr('dtype'))

File: python/paddle/incubate/autograd/primrules.py [hunk 2]
```diff
@@ -464,6 +465,38 @@ def dropout_orig2prim(op, seed_t, x):
...
+                              shape=None,
+                              shape_t=shape_t,
+                              shape_tl=None)
+    return uniform_random(dtype,
+                          min_value,
+                          max_value,
+                          seed,
+                          shape=op.attr('shape'),
+                          shape_t=None,
+                          shape_tl=None)
```
Comment (cxxly) [inline python/paddle/incubate/autograd/primrules.py]: 此处不用写三个分支，shape_tl/shape_t不存在同时不为None的情况，直接返回就行
```python
    return uniform_random(dtype,
                          min_value,
                          max_value,
                          seed,
                          shape=op.attr('shape'),
                          shape_t=shape_t,
                          shape_tl=shape_tl)
```

File: paddle/fluid/operators/prim_ops/uniform_random_p_op.cc [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    AddComment(R"DOC(
+Autograd primitive uniform_random_p operator.
+)DOC");
+  }
+};
+
+class UniformRandomPrimOpShapeInference : public framework::InferShapeBase {
+ public:
+  void operator()(framework::InferShapeContext *ctx) const override {
+    // framework::InferShapeVarPtr y_var_ptr = ctx->GetOutputVarPtrs("Y")[0];
```
Comment (JiabinYang) [inline paddle/fluid/operators/prim_ops/uniform_random_p_op.cc]: remove useless code

File: paddle/fluid/operators/prim_ops/uniform_random_p_op.cc [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+               const platform::Place &dev_place) const override {
+    PADDLE_THROW(platform::errors::Unimplemented(
+        "Prim operator uniform_randrom_p should not be excuted directly"));
+  }
+};
+
+class UniformRandomPrimOpMaker : public framework::OpProtoAndCheckerMaker {
+ public:
+  void Make() override {
+    AddInput("ShapeTensor",
```
Comment (JiabinYang) [inline paddle/fluid/operators/prim_ops/uniform_random_p_op.cc]: Will this cause dynamic shape error

**Review by JiabinYang — COMMENTED** [review]
some comments

File: python/paddle/incubate/autograd/primrules.py (hunk 1 above)
Comment (Charles-hit) [reply python/paddle/incubate/autograd/primrules.py]: done

File: python/paddle/incubate/autograd/primrules.py (hunk 2 above)
Comment (Charles-hit) [reply python/paddle/incubate/autograd/primrules.py]: done

File: paddle/fluid/operators/prim_ops/uniform_random_p_op.cc (hunk 3 above)
Comment (Charles-hit) [reply paddle/fluid/operators/prim_ops/uniform_random_p_op.cc]: done

File: paddle/fluid/operators/prim_ops/uniform_random_p_op.cc (hunk 4 above)
Comment (Charles-hit) [reply paddle/fluid/operators/prim_ops/uniform_random_p_op.cc]: yes,it has been deleted.

**Review by cxxly — APPROVED** [review]
LGTM

**Review by JiabinYang — APPROVED** [review]
LGTM

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**